lib/hutch/publisher.rb

```diff
@@ -41,16 +41,15 @@ def publish(routing_key, message, properties = {}, options = {})
 
     private
 
-    def log_publication(serializer, payload, routing_key)
-      logger.info {
-        spec =
-          if serializer.binary?
-            "#{payload.bytesize} bytes message"
-          else
-            "message '#{payload}'"
-          end
-        "publishing #{spec} to #{routing_key}"
-      }
+    def log_publication(serializer, payload, routing_key, message_id)
+      logger.info("publishing message with id: #{message_id} to #{routing_key}")
+
+      spec = if serializer.binary?
+        "publishing a #{payload.bytesize} bytes message"
+      else
+        "publishing message: '#{payload}'"
+      end
+      logger.debug(spec)
```

Comment:
💭 It looks like the original code used the block-variant of `logger.info`. According to [the `Logger` docs](https://ruby-doc.org/stdlib-2.4.0/libdoc/logger/rdoc/Logger.html#class-Logger-label-How+to+log+a+message), that has lazy evaluation.

Would it be worth to use the block-type `logger.debug { ... }` to avoid evaluating the `spec =` on every log level?